paymentsheet/src/main/java/com/stripe/android/paymentsheet/forms/PaymentMethodCommonSpec.kt

```diff
@@ -1,38 +1,46 @@
 package com.stripe.android.paymentsheet.forms
 
+import androidx.annotation.StringRes
+import androidx.compose.ui.graphics.Color
+import com.stripe.android.paymentsheet.forms.FormElementSpec.SectionSpec
+
 
 /**
  * This class is used to define different forms full of fields.
  */
 data class FormSpec(
-    val layout: FieldLayoutSpec,
+    val layout: Layout,
     val paramKey: MutableMap<String, Any?>,
-) {
-    val allTypes get() = layout.sections.map { it.field }
-}
+)
 
 /**
  * This is a data representation of the layout of UI fields on the screen.
  */
-data class FieldLayoutSpec(val sections: List<SectionSpec>) {
-    val allFields get() = sections.map { it.field }
+data class Layout(val elements: List<FormElementSpec>) {
+    val allFields
+        get() = elements.filterIsInstance<SectionSpec>().map { it.field }
```

Comment:
is `filterIsInstance` required? If we want to add a Layout that only renders `SectionSpec`, I guess we should create a `SectionSpecLayout` class that only takes a `List<SectionSpec>`

Reply:
Fields have a particular meaning here, it is specifically a place where a user can enter a value.  Fields are only allowed to be in sections, so that it why we only look at the section specs.   Does this answer your question?